### PR TITLE
JBPM-10162 IntermediateThrowingSignal node from subprocess and the su…

### PR DESCRIPTION
**Pull request by bxf12315**
…bprocess is not getting marked as executed

**Thank you for submitting this pull request**

**JIRA**: _(please edit the JIRA link if it exists)_ 

[https://issues.redhat.com/browse/JBPM-10162](https://issues.redhat.com/browse/JBPM-10162)

**referenced Pull Requests**: _(please edit the URLs of referenced pullrequests if they exist)_

* paste the link(s) from GitHub here
* link 2
* link 3 etc.

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Comment by bxf12315** [comment]
Jenkins run fdb

**Comment by bxf12315** [comment]
Jenkins run fdb

**Comment by bxf12315** [comment]
Jenkins retest this

**Comment by fjtirado** [comment]
@bxf12315 Can you link this PR (with is just testing) to the PR which fixes the issue?

**Comment by gmunozfe** [comment]
@bxf12315 this change breaks backward compatibility (notice that one test had to be modified). Of course, it can be argued that the current behavior was missing these states (obsolete, skipped) inside `completed` because the only alternative is to consider them as `active` and a new API is probably a complex solution at this moment. 
Do you know if this API is used anywhere else from the UI?

**Comment by bxf12315** [comment]
> @bxf12315 this change breaks backward compatibility (notice that one test had to be modified). Of course, it can be argued that the current behavior was missing these states (obsolete, skipped) inside `completed` because the only alternative is to consider them as `active` and a new API is probably a complex solution at this moment. Do you know if this API is used anywhere else from the UI?

The method is just called by [1], what do you think? 

[1] https://github.com/Cory-jbpm/droolsjbpm-integration/blob/41e2c6f493b8e15a0f358334927fb4dea8967dcd/kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/ImageServiceBase.java#L124

**Comment by gmunozfe** [comment]
> > @bxf12315 this change breaks backward compatibility (notice that one test had to be modified). Of course, it can be argued that the current behavior was missing these states (obsolete, skipped) inside `completed` because the only alternative is to consider them as `active` and a new API is probably a complex solution at this moment. Do you know if this API is used anywhere else from the UI?
> 
> The method is just called by [1], what do you think?
> 
> [1] https://github.com/Cory-jbpm/droolsjbpm-integration/blob/41e2c6f493b8e15a0f358334927fb4dea8967dcd/kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/ImageServiceBase.java#L124

ok, thanks for clarification @bxf12315

File: jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/RuntimeDataServiceImpl.java [hunk 1]
```diff
@@ -692,6 +692,13 @@ public Collection<NodeInstanceDesc> getProcessInstanceHistoryCompleted(long proc
         return getProcessInstanceHistory(processId, true, queryContext);
     }
 
+    @Override
+    public Collection<NodeInstanceDesc> getProcessInstanceHistoryPassed(long processId, QueryContext queryContext) {
+        Map<String, Object> params = new HashMap<String, Object>();
+        params.put("processId", processId);
```
Comment (fjtirado) [inline jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/RuntimeDataServiceImpl.java:698]:
```suggestion
        Map<String, Object> params = Collections.singletonMap("processId", processId);
```

**Comment by bxf12315** [comment]
Jenkins retest this

**Comment by bxf12315** [comment]
Jenkins retest this

**Review by gmunozfe — APPROVED** [review]
LGTM, well done @bxf12315 !

**Pull request merged**